backend/httpclient/provider.go

```diff
@@ -161,12 +172,24 @@ func (p *Provider) createClientOptions(providedOpts ...Options) Options {
 		clientOpts = providedOpts[0]
 		clientOpts.Middlewares = append(clientOpts.Middlewares, p.Opts.Middlewares...)
 	default:
-		panic("only an empty or one Options is valid as argument")
+		return Options{}, errors.New("only an empty or one Options is valid as argument")
+	}
+
+	// Verify that middleware names are unique
+	middlewareNames := make(map[string]struct{})
+	for _, m := range clientOpts.Middlewares {
+		nm, ok := m.(MiddlewareName)
+		if ok {
+			if _, exists := middlewareNames[nm.MiddlewareName()]; exists {
+				return Options{}, fmt.Errorf("middleware with name %s already exists", nm.MiddlewareName())
+			}
+			middlewareNames[nm.MiddlewareName()] = struct{}{}
+		}
```

Comment:
Configure Middleware below can change Middlewares in theory so we probably need to validate after that func had been called